src/parser/jobs/blm/modules/Sharpcast.js

```diff
@@ -0,0 +1,136 @@
+import {getDataBy} from 'data'
+import ACTIONS from 'data/ACTIONS'
+import STATUSES from 'data/STATUSES'
+import Module from 'parser/core/Module'
+import React from 'react'
+import {Item} from 'parser/core/modules/Timeline'
+import {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+import {Trans, Plural} from '@lingui/react'
+import {StatusLink} from 'components/ui/DbLink'
+
+const SHARPCAST_DURATION_MILLIS = STATUSES.SHARPCAST.duration * 1000
+
+const SHARPCAST_CONSUMER_IDS = [
+	ACTIONS.FIRE_I.id,
+	ACTIONS.THUNDER_III.id,
+	ACTIONS.THUNDER_IV.id,
+	ACTIONS.SCATHE.id,
+]
+
+export default class Sharpcast extends Module {
+	static handle = 'sharpcast'
+	static dependencies = [
+		'timeline',
+		'procs', // eslint-disable-line @xivanalysis/no-unused-dependencies
+		'suggestions',
+	]
+
+	_buffWindows = {
+		current: null,
+		history: [],
+	}
+
+	_droppedSharpcasts = 0
+
+	constructor(...args) {
+		super(...args)
+		this.addHook('removebuff', {by: 'player', abilityId: STATUSES.SHARPCAST.id}, this._onRemoveSharpcast)
+		this.addHook('applybuff', {by: 'player', abilityId: STATUSES.SHARPCAST.id}, this._onGainSharpcast)
+		this.addHook('cast', {by: 'player', abilityId: SHARPCAST_CONSUMER_IDS}, this._onCast)
+		this.addHook('death', {to: 'player'}, this._onDeath)
+		this.addHook('complete', this._onComplete)
+	}
+
+	_onRemoveSharpcast(event) {
+		this._stopAndSave(event.timestamp)
+	}
+
+	_onGainSharpcast(event) {
```

Comment:
It may be simpler and cleaner to use `this.combatants.selected.hasStatus()` instead of tracking buff windows manually; you can look at [Kassatsu.js](https://github.com/xivanalysis/xivanalysis/blob/master/src/parser/jobs/nin/modules/Kassatsu.js#L41) for an example.

Reply:
I'm using the buff window data to show it on the timeline, so I'm going to need to keep it anyways. If I wasn't trying to do that, I agree, your idea would be cleaner.